addons/point_of_sale/upgrades/1.0.2/post-deduplicate-uuids.py

```diff
@@ -20,18 +20,21 @@ def migrate(cr, version):
     def deduplicate_uuids(table):
         cr.execute(
             f"""
-                SELECT UNNEST(ARRAY_AGG(id))
+                SELECT id
                   FROM {table}
-              GROUP BY uuid
+              GROUP BY id, uuid
                 HAVING COUNT(*) > 1
             """
         )
+        while True:
+            batch = cr.fetchmany(100000)
```

Comment:
This looks incorrect. You are making extra queries between each fetchmany. What you can do instead is to run the select multiple times until nothing is selected:
```suggestion
       query = f"""
          SELECT id
            FROM {table}
        GROUP BY id, uuid
          HAVING COUNT(*) > 1
       """
       while True:
           cr.execute(query)
           if not cr.rowcount:
               break
           ids = [r[0] for r in fetchmany(100000)]
           cr.execute(
                f"UPDATE {table} SET uuid = (%s::json)->>(id::text) WHERE id IN %s",
                [Json({id_: str(uuid.uuid4()) for id_ in ids}), tuple(ids)]
           )   
```